### DISCUSSION TITLE
don't ready kafkatopic when I use kraft mode

### DISCUSSION THREAD

**Discussion by fly82** [strimzi/strimzi-kafka-operator#8142] [General]
kubectl create ns kafka
kubectl create -f 'https://strimzi.io/install/latest?namespace=kafka' -n kafka
kubectl -n kafka set env deployment/strimzi-cluster-operator STRIMZI_FEATURE_GATES=+UseKRaft
simple kafka manifest and topic
result:
kubectl get kt -n kafka
NAME           CLUSTER      PARTITIONS   REPLICATION FACTOR   READY
example-demo   my-cluster   3            3

**Comment by scholzj** [comment]
Topic Operator is not supported in KRaft mode currently. So the `KafkaTopic! resources are basically ignored. So this is expected behavior.

**Reply by fly82** [reply]
Do You have the plan to support Topic Operator in KRaft mode?

**Reply by scholzj** [reply]
We would love to. But it is not completely trivial as KRaft does not offer some of the features which ZooKeeper was providing (such as watching for changes). That makes it quite complicated to implement it in KRaft mode. We are looking into it right now, but it remains to be seen if it is possible with some reasonable amount of effort or not.
